File: app/javascript/channels/room_channel.js
import consumer from "channels/consumer"

// noinspection JSUnresolvedVariable
const appRoom = consumer.subscriptions.create("RoomChannel", {
  connected() {
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    return alert(JSON.stringify(data, null, "  "));
  },

  speak: function (message) {
    // noinspection JSUnresolvedFunction
    return this.perform('speak', { message: message });
  },
});

// noinspection JSUnresolvedVariable
window.document.onkeydown = function (event) {
  // noinspection JSUnresolvedVariable
  if (event.key === 'Enter' && event.target.value) {
    console.log(event.target.value)
    appRoom.speak(event.target.value);
    event.target.value = '';
    event.preventDefault();
  }
}

File: app/channels/room_channel.rb
# frozen_string_literal: true

class RoomChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from "room_#{params[:room_uuid]}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    ActionCable.server.broadcast("room_#{params[:room_uuid]}", { message: data['message'] })
  end
end
